File: src/main/scala/com/fp/distractor/core/transport/telnet/TelnetHandler.java
package com.fp.distractor.core.transport.telnet;

import akka.actor.ActorRef;
import akka.actor.ActorSelection;
import akka.event.LoggingAdapter;
import com.fp.distractor.core.api.DistractorApi;
import org.apache.mina.core.service.IoHandlerAdapter;
import org.apache.mina.core.session.IoSession;

public class TelnetHandler extends IoHandlerAdapter {

    private final LoggingAdapter log;
    private final ActorSelection distractor;
    private final ActorRef telnetTransport;

    public TelnetHandler(LoggingAdapter log, ActorSelection distractor, ActorRef telnetTransport) {
        this.log = log;
        this.distractor = distractor;
        this.telnetTransport = telnetTransport;
    }

    @Override
    public void messageReceived(IoSession session, Object message) throws Exception {
        distractor.tell(new DistractorApi.DistractorRequest((String) message), telnetTransport);
    }
}

File: src/main/scala/com/gmaslowski/distractor/transport/telnet/TelnetHandler.scala
package com.gmaslowski.distractor.transport.telnet

import akka.actor.{ActorRef, ActorSelection}
import akka.event.LoggingAdapter
import com.gmaslowski.distractor.core.api.DistractorApi
import org.apache.mina.core.service.IoHandlerAdapter
import org.apache.mina.core.session.IoSession

class TelnetHandler(val log: LoggingAdapter, val distractor: ActorSelection, val telnetTransport: ActorRef) extends IoHandlerAdapter {

  @throws(classOf[Exception])
  override def messageReceived(session: IoSession, message: AnyRef) {
    distractor.tell(new DistractorApi.DistractorRequest(message.asInstanceOf[String]), telnetTransport)
  }
}